beam/public/js/scan/scan.js

```diff
@@ -208,6 +208,24 @@ class ScanHandler {
 				frappe.model.set_value(row.doctype, row.name, 's_warehouse', barcode_context.target)
 				frappe.model.set_value(row.doctype, row.name, 't_warehouse', barcode_context.target)
 			}
+		} else if (barcode_context.doctype == 'Stock Reconciliation Item') {
+			frappe.model.set_value(cur_frm.doc.doctype, cur_frm.doc.name, 'set_warehouse', barcode_context.context.doc.name)
```

Comment:
If you're setting a value in the parent document, then you can re-use the `cur_frm` APIs:

```suggestion
			cur_frm.set_value('set_warehouse', barcode_context.context.doc.name)
```